code/modules/antagonists/changeling/powers/spiders.dm

```diff
@@ -1,18 +1,145 @@
+	if(do_after(user, 4 SECONDS, FALSE, target = user)) // Takes 5 seconds to spawn a spider
+		spider_counter++
+		user.visible_message("<span class='danger'>[user] vomits an arachnid!</span>")
+		var/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/S = new(user.loc)
+		S.owner_UID = user.UID()
+		S.faction |= list("spiders", "\ref[owner]") // Makes them friendly only to the owner & other spiders
+		SSblackbox.record_feedback("nested tally", "changeling_powers", 1, list("[name]"))
+		is_operating = FALSE
+		return TRUE
+	is_operating = FALSE
+	return FALSE
 
-	SSblackbox.record_feedback("nested tally", "changeling_powers", 1, list("[name]"))
-	return TRUE
+// Child of giant_spider because this should do everything the spider does and more
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider
+	var/mob/owner_UID // References to the owner changeling
+	var/gibbed = FALSE // To check if the spider died via gib or if it still needs to be gibbed, to prevent post-death revival
+	var/currentOrder = 0
+	var/list/enemies = list() // BIG AAAA TEST FOR RETALIATE
+	venom_per_bite = 3
+	speak_chance = 0
+	wander = 0
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/gib()
+	gibbed = TRUE
+	return ..()
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/death()
+	if(!gibbed)
+		gib()
+		return
+	var/mob/owner_mob = locateUID(owner_UID)
+	var/datum/action/changeling/spiders/S = locate() in owner_mob.actions
+	S.spider_counter--
+	return ..()
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/examine(mob/user)
+	. = ..()
+	if(user.UID() == owner_UID)
+		. += "<span class='notice'>It is currently angy!</span>"
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/AltShiftClick(mob/user)
+	. = ..()
+	if(user.UID() == owner_UID)
+		spider_order(user)
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/proc/spider_order(mob/user)
+	switch(currentOrder)
+		if(0)
+			to_chat(user, "<span class='notice'>We order the giant spider to follow us but attack anything on sight.</span>")
+			currentOrder = 1
+		if(1)
+			to_chat(user, "<span class='notice'>We order the giant spider to follow us and stay calm, only attacking if we or it are attacked.</span>")
+			currentOrder = 2
+		if(2)
+			to_chat(user, "<span class='notice'>We order the giant spider to remain idle and calm, only attacking if we or it are attacked.</span>")
+			currentOrder = 3
+		if(3)
+			to_chat(user, "<span class='notice'>We order the giant spider to remain idle but attack anything on sight.</span>")
+			currentOrder = 0
+	handle_automated_movement()
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/handle_automated_movement() //Hacky and ugly.
+	. = ..()
+	var/list/around = view(src, vision_range)
```

Comment:
You should probably declare this only when you need it. It can get expensive, and it's possible we get everything in view and then just return and not even use it.

Reply:
Yeah; All except IDLE_RETALIATE uses it, and at the end of even it, there is an around check for other spiders so they are able to all retaliate as a team. I'll put this on resolve, seeing as putting it under specific ones would make Idle_Retaliate also not do groupal retaliation.